File: Backend/Models/gestationAge.js
import mongoose from 'mongoose';
const gestAge = new mongoose.Schema({
    screeningId:{
        type: String,
        required: true
    },
    lmp:{
        type: Date,
        required: true,
        default: null
    },
    ultrasoundDate:{
        usWeeks:{
            type: Number,
            required: true
        },
        usDays:{
            type: Number,
            required: true
        }
    },
    lmpCertainty:{
        type: Boolean,
        required: true
    },
    enrolmentDate:{
        type: Date,
        required:true
    }

});

const gestationAge = mongoose.model('gestationAge', gestAge);

export default gestationAge;

File: api/Models/GestationAge.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace KemriApi.Models
{
    public class GestationAge
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? Id { get; set; }

        [BsonElement("screeningId")]
        public string ScreeningId { get; set; } = string.Empty;

        [BsonElement("lmp")]
        public DateTime? Lmp { get; set; }

        [BsonElement("ultrasoundDate")]
        public UltrasoundDateInfo UltrasoundDate { get; set; } = new();

        [BsonElement("lmpCertainty")]
        public string LmpCertainty { get; set; } = string.Empty;

        [BsonElement("enrolmentDate")]
        public DateTime EnrolmentDate { get; set; }

        [BsonElement("estDueDate")]
        public DateTime EstDueDate { get; set; }

        [BsonElement("currentGestAge")]
        public CurrentGestAgeInfo CurrentGestAge { get; set; } = new();
    }

    public class UltrasoundDateInfo
    {
        [BsonElement("usWeeks")]
        public int UsWeeks { get; set; }

        [BsonElement("usDays")]
        public int UsDays { get; set; }
    }

    public class CurrentGestAgeInfo
    {
        [BsonElement("gestweeks")]
        public int GestWeeks { get; set; }

        [BsonElement("gestdays")]
        public int GestDays { get; set; }
    }
}
